core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -315,6 +315,12 @@ package object config {
     .bytesConf(ByteUnit.MiB)
     .createOptional
 
+  private[spark] val ENABLE_EXECUTOR_TREE_AGGREGATE = ConfigBuilder("spark.executor.treeAggregate")
```

Comment:
If we will add it as a configuration, I would rename it something like `spark.rdd.treeAggregate.finalAggregateOnExecutor` (feel free to change the wording).

Reply:
BTW, let's don't forget to add `since` version too